### DISCUSSION TITLE
NativeAOT Support

### DISCUSSION THREAD

**Discussion by 4OTbIPE** [microsoft/WindowsAppSDK#3856] [Dev team discussions]
Is there any news on NativeAOT support for WindowsAppSDK please? Progress? Maybe approximate dates? The lack of NativeAOT is probably the only block for me to switch from UWP(

**Comment by ghost1372** [comment]
> Is there any news on NativeAOT support for WindowsAppSDK please? Progress? Maybe approximate dates? The lack of NativeAOT is probably the only block for me to switch from UWP(

Net 8.0 only support console and Asp app, mybe we should wait for Net 9.0

**Reply by Wufus** [reply]
It is not supported in .Net 9 yet

**Reply by ghost1372** [reply]
> It is not supported in .Net 9 yet

According to the [roadmap](https://github.com/microsoft/WindowsAppSDK/discussions/4256#discussion-6332577), it will be supported in version 1.6 
Therefore, it will be added to .NET 9 soon.  I think

**Comment by manodasanW** [comment]
WinAppSDK is dependent on [CsWinRT](https://github.com/microsoft/CsWinRT) being AOT compatible.  We have been making progress on the CsWinRT side in making our generated projections AOT compatible.  We had a couple PRs merged with respect to that and with some larger ones related to AOT compat issues in our generic collections support upcoming very soon.  One of the merged PRs was in the .NET repo to make sure the .NET AOT runtime supported a couple things that WinAppSDK was going to need and that made it for .NET 8.

Once CsWinRT is AOT compatible, we will be evaluating where it stands with WinAppSDK and going from there.  We will keep folks posted as we make progress.

**Reply by 4OTbIPE** [reply]
Thanks a lot for the answer!

**Reply by strich** [reply]
@manodasanW how are things going on this front? Is there still a long road ahead? Wondering if it can all come in for the upcoming .NET 8 release.

**Reply by jingliancui** [reply]
Hi, how are you , .Net 8 was release , is there any progress on aot? thanks.

**Reply by ghost** [reply]
Looking forward to more AOT support for destkop applications :)

**Reply by beeradmoore** [reply]
Hey @manodasanW , where is the best place to keep updated? Can we get an issue up we can subscribe to, or should we just watch this thread?

**Reply by driver1998** [reply]
@beeradmoore You can see the `staging/AOT` branch and PRs going towards that. A lot of stuff is working now.

**Comment by michael-hawker** [comment]
The latest 1.6-preview1 build has the initial support for AOT compilation, see the release notes here: https://learn.microsoft.com/windows/apps/windows-app-sdk/preview-channel#native-aot-support

**Comment by Symbai** [comment]
2025, whats the status? Can we use .NET + AOT + XAML?

**Reply by driver1998** [reply]
It should work just fine now.

**Reply by ghost1372** [reply]
Its working fine. But xaml and binding needs some improvments
